### PR TITLE
Manage swap partition as a hard requirement

### PR DESCRIPTION
**Pull request by bogdando**
Add swap partition management into the bootstrap role

The swap file/partition management logic is carried over from the existing EDPM pre-adoption implementation, but moved out of the nova compute specific scope into generic tasks for EDPM.

Depends-On: https://review.rdoproject.org/r/c/rdo-jobs/+/48683

Closes: [OSPRH-133](https://issues.redhat.com//browse/OSPRH-133)

### CODE REVIEW

**Comment by bogdando** [comment]
The molecule testing is blocked as we cannot use ``swapon`` from podman containers, we'd need a delegated driver for that, likely. Any hints to make this unblocked, @raukadah perchance?

**Comment by bogdando** [comment]
the rdo-project job is failed, probably by unrelated reason, because I can see the swap task completed there

https://logserver.rdoproject.org/56/156/814b11000ae0869a6e1559bee243ec11f332ff28/github-check/edpm-ansible-github-rdo-integration-centos-8-crc-singlenode-centos-9-external-compute/79eb0cc/controller/controller/pod/deploy-external-dataplane-compute-n4zc8-logs.txt

**Comment by bogdando** [comment]
check-rdo

**Comment by bogdando** [comment]
check-rdo

**Comment by jpodivin** [comment]
Just to clarify. The molecule in podman seems to work with this patch after all https://github.com/openstack-k8s-operators/edpm-ansible/actions/runs/5014407281/jobs/8988616358?pr=156 . Do you still want to replace it with delegated driver?

File: roles/edpm_bootstrap/defaults/main.yml
Comment (kajinamit) [inline roles/edpm_bootstrap/defaults/main.yml:57]: I think we should hard-fail if the partition with the given label is not found.

File: roles/edpm_bootstrap/defaults/main.yml
Comment (kajinamit) [inline roles/edpm_bootstrap/defaults/main.yml:56]: The var name does not represent well that this is a boolean flag to enable swap. Probably edpm_bootstrap_enable_swap (or swap_enabled) is better.

File: roles/edpm_bootstrap/tasks/bootstrap.yml
Comment (kajinamit) [inline roles/edpm_bootstrap/tasks/bootstrap.yml:177]: Can't we use the native module to manipulate fstab entries ?

File: roles/edpm_bootstrap/tasks/bootstrap.yml
Comment (bogdando) [reply roles/edpm_bootstrap/tasks/bootstrap.yml:164]: According to the original [tripleo design spec](https://opendev.org/openstack/tripleo-specs/commit/59e352f1a928c8469cb46756eac6113397d06de4) which established the standalone roles aka EDPM for NG, there are the following tasks (stages):

* bootstrap
* install
* pre-network
* network
* configure
* container-config
* service-bootstrap

While those are likely a subject of changes, the idea behind, I believe is to reduce the need for future migrations, like using mutating hooks to update all CRs for EDPM objects, when changing the tasks and roles (if we expose it as a public/partners integration interface)

So, no, we should keep it in the bootstrap.yaml tasks

File: roles/edpm_bootstrap/tasks/bootstrap.yml
Comment (bogdando) [reply roles/edpm_bootstrap/tasks/bootstrap.yml:177]: Those are valid points, just please note that this change ports the unmodified implementation that we used to have in tripleo, and still do have in the layered product (where this feature is a release blocker for the upcoming release relying now on EDPM). This implementation has been suppored. We need to unblock the upcoming release by porting it, then we can think on changing it

File: roles/edpm_bootstrap/tasks/bootstrap.yml
Comment (slagle) [reply roles/edpm_bootstrap/tasks/bootstrap.yml:164]: I don't have any strong opinion on splitting it out into a separate role. A separate task file would be fine though that could just be included here. It might make things a bit cleaner, and adds the ability to execute just the swap tasks if needed.

File: roles/edpm_bootstrap/defaults/main.yml
Comment (bogdando) [reply roles/edpm_bootstrap/defaults/main.yml:57]: There is a fallback path https://github.com/bogdando/edpm-ansible/blob/f6144dd7cc0d767d2469a8922d93042a92bbaeaf/roles/edpm_bootstrap/tasks/swap.yml#L50 in such a case.
Please rememver, this is a direct port of the tripleo supported feature. We can adjust it in followups

**Comment by bogdando** [comment]
Ci failure looks unrelated https://github.com/openstack-k8s-operators/edpm-ansible/actions/runs/5199776741/jobs/9377637527#step:6:249

File: roles/edpm_bootstrap/tasks/swap.yml
Comment (bshephar) [inline roles/edpm_bootstrap/tasks/swap.yml:22]: We should probably use `ansible.builtin.shell` for these. I think we were trying to get consistency around the FQCN's with this project.

File: roles/edpm_bootstrap/tasks/swap.yml
Comment (bshephar) [inline roles/edpm_bootstrap/tasks/swap.yml:39]: And here `ansible.builtin.shell`.

File: roles/edpm_bootstrap/tasks/swap.yml
Comment (bshephar) [inline roles/edpm_bootstrap/tasks/swap.yml:32]: I see this has already been discussed, but it would be nice if we can follow up after this is merged to reimplement using: https://docs.ansible.com/ansible/latest/collections/ansible/posix/mount_module.html

**Review by bshephar — COMMENTED** [review]
I can see this role has some inconsistencies with FQCN usage. But I think we should probably aim for using the FQCN's with these new PR's. We can clean up the rest of the role later.

I see you mentioned in the comments that you're just porting a pre-existing thing to edpm-ansible in order to unblock some work. So I'm happy to merge this one and address the comments I made in a follow up PR if you prefer.

**Comment by bogdando** [comment]
@kajinamit PTAL

File: roles/edpm_bootstrap/tasks/swap.yml
Comment (bogdando) [reply roles/edpm_bootstrap/tasks/swap.yml:22]: ack, let's keep this for follow-ups

File: roles/edpm_bootstrap/tasks/swap.yml
Comment (bogdando) [reply roles/edpm_bootstrap/tasks/swap.yml:32]: ack, let's keep this for follow-ups

File: roles/edpm_bootstrap/tasks/swap.yml
Comment (bogdando) [reply roles/edpm_bootstrap/tasks/swap.yml:39]: ack, let's keep this for follow-ups

**Review by kajinamit — APPROVED** [review]
I don't know why github doesn't remove my change request.

**Comment by rebtoor** [comment]
recheck

**Pull request merged**
